elb_access_logs_bucket/main.tf

```diff
@@ -94,67 +115,70 @@ resource "aws_s3_bucket" "logs" {
   ]
 }
 EOF
-
-
-tags = {
-  Environment = "All"
 }
 
-# In theory we should only put one copy of every file, so I don't think this
-# will increase space, just give us history in case we accidentally
-# delete/modify something.
-versioning {
-  enabled = true
-}
+resource "aws_s3_bucket_versioning" "logs" {
+  bucket = aws_s3_bucket.logs.id
 
-server_side_encryption_configuration {
-  rule {
-    apply_server_side_encryption_by_default {
-      sse_algorithm = "aws:kms"
-    }
+  # In theory we should only put one copy of every file, so I don't think this
+  # will increase space, just give us history in case we accidentally
+  # delete/modify something.
+  versioning_configuration {
+    status = "Enabled"
   }
 }
 
-# Lifecycle rules: configure a sliding window for moving logs from standard
-# storage to standard Infrequent Access, then to glacier, then deleting them.
-# The rules will only be enabled if the lifecycle day threshold is set to a
-# positive number.
+resource "aws_s3_bucket_lifecycle_configuration" "logs" {
+  bucket = aws_s3_bucket.logs.id
 
-lifecycle_rule {
-  id      = "log_aging_ia"
-  enabled = var.lifecycle_days_standard_ia > 0 ? true : false
+  # Lifecycle rules: configure a sliding window for moving logs from standard
+  # storage to standard Infrequent Access, then to glacier, then deleting them.
+  # The rules will only be enabled if the lifecycle day threshold is set to a
+  # positive number.
 
-  prefix = "/"
+  rule {
+    id     = "log_aging_ia"
+    status = var.lifecycle_days_standard_ia > 0 ? "Enabled" : "Disabled"
+
+    filter {
+      prefix = "/"
+    }
 
-  transition {
-    days          = var.lifecycle_days_standard_ia
-    storage_class = "STANDARD_IA"
+    transition {
+      days          = var.lifecycle_days_standard_ia
+      storage_class = "STANDARD_IA"
+    }
   }
-}
 
-lifecycle_rule {
-  id      = "log_aging_glacier"
-  enabled = var.lifecycle_days_glacier > 0 ? true : false
+  rule {
+    id     = "log_aging_glacier"
```

Comment:
I am tempted to ban regular Glacier use.   The storage savings can not make up for the pain and cost if we ever need to pull the data.    Standard-IA makes more sense for fixed lifecycle use or Intelligent Tiering for buckets where lifecycle is not clear.

Reply:
Ack, I totally missed this! To add in a future version, probably.